pkg/assetFetcher/errors.go

```diff
@@ -0,0 +1,47 @@
+package assetfetcher
+
+import "fmt"
+
+// URLParsingError when trying to parse the given host or URL
+type URLParsingError struct {
+	Inner      error
+	NexposeURL string
+}
+
+// Error returns a URLParsingError
+func (e *URLParsingError) Error() string {
+	return fmt.Sprintf("Error parsing Nexpose URL (%v): %v", e.NexposeURL, e.Inner.Error())
+}
+
+// ResponseParsingError when trying to parse a Nexpose response
+type ResponseParsingError struct {
+	Inner      error
+	NexposeURL string
+}
+
+// Error returns a ResponseParsingError
+func (e *ResponseParsingError) Error() string {
+	return fmt.Sprintf("Error parsing Nexpose response from %v: %v", e.NexposeURL, e.Inner.Error())
+}
+
+// ErrorReadingNexposeResponse represents an error returned when the response from Nexpose can't be read
+type ErrorReadingNexposeResponse struct {
```

Comment:
What's the difference between this and the `ResponseParsingError`?

Reply:
`ErrorReadingNexposeResponse` is the error you will get when Nexpose sends a response, but we can't read it (ioutil.ReadAll fails), `ResponseParsingError` is the error you get if the response _can_ be read, but _can't_ be parsed into our `SiteAssetsResponse` struct. Would a more specific name for `ResponseParsingError` help, like maybe `ErrorParsingJSONResponse` to indicate that it's likely an issue with the JSON they're sending back?

Reply:
Yes, I think that'd be helpful